File: test-suite-kotlin/src/test/kotlin/io/micronaut/chatbots/telegram/AboutCommandHandler.kt
package io.micronaut.chatbots.telegram

import io.micronaut.chatbots.core.SpaceParser
import io.micronaut.chatbots.core.TextResourceLoader
import io.micronaut.chatbots.telegram.api.Chat
import io.micronaut.chatbots.telegram.api.Update
import io.micronaut.chatbots.telegram.core.CommandHandler
import io.micronaut.chatbots.telegram.core.TelegramSlashCommandParser
import io.micronaut.context.annotation.Requires
import jakarta.inject.Singleton

@Requires(property = "spec.name", value = "telegram")
// tag::class[]
@Singleton
class AboutCommandHandler(
    slashCommandParser: TelegramSlashCommandParser,
    textResourceLoader: TextResourceLoader,
    spaceParser: SpaceParser<Update, Chat>
) : CommandHandler(slashCommandParser, textResourceLoader, spaceParser) {

    override fun getCommand() = COMMAND_ABOUT

    companion object {
        private const val COMMAND_ABOUT = "/about"
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/chatbots/docs/telegram/AboutCommandHandler.py
from jakarta.inject import Singleton
from micronaut.chatbots.core import SpaceParser, TextResourceLoader
from micronaut.chatbots.telegram.api import Chat, Update
from micronaut.chatbots.telegram.core import CommandHandler, TelegramSlashCommandParser
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="telegram")
# tag::class[]
@Singleton
class AboutCommandHandler(CommandHandler):

    COMMAND_ABOUT: str = "/about"

    def __init__(self,
                 slash_command_parser: TelegramSlashCommandParser,
                 text_resource_loader: TextResourceLoader,
                 space_parser: SpaceParser[Update, Chat]):
        super().__init__(slash_command_parser, text_resource_loader, space_parser)

    def getCommand(self) -> str:
        return AboutCommandHandler.COMMAND_ABOUT
# end::class[]
